### DISCUSSION TITLE
Question about EuPI

### DISCUSSION THREAD

**Discussion by SeonJK** [euphony-io/euphony#182] [Q&A] [answered]
I'd like to adapt EuPI for our project; order-with.

In our project,
I implemented and checked `callEuPI()` method for TxManager

But I have some problem on RxManager.
When I use EuPI like:
```kotlin
val rxManager: EuRxManager() = EuRxManager(EuOption.ModeType. EUPI)

if (rxManager.listen()) {
  Log.d(TAG, "rxManager listen return : true")
} else {
  Log.d(TAG, "rxManager listen return : false")
}

mEuPIRxManager.setOnWaveKeyPressed(RequestCodeEnum.MENU_REQUEST.code.toInt()) {  // 18500.0 -> 18500
  Toast.makeText(this, "menu request detected", Toast.LENGTH_SHORT).show()
}
```
But It doesn't work and in Log:

> D/[StoreActivity]: StoreActivity - mEuPIRxManager listen return : false

<br/><br/>

I modify some code to initialize rxManager like:
```kotlin
val rxManager: EuRxManager() = EuRxManager()
```

app also speak the log but rxManager can listen data with no toast messages!!

> D/[StoreActivity]: StoreActivity - mEuPIRxManager listen return : false

<br/><br/><br/><br/>

My Questions are...
- When I initialize `rxManager(EuOption.ModeType. EUPI)`, any further codes to need?
- `setOnWaveKeyPressesd`, `setOnWaveKeyUp`, `setOnWaveKeyDown` need parameter `int key`, and `callEuPI` needs `double frequency`. is the cause about my problem?

Thanks in advance :)

**Comment by designe** [comment, marked as answer]
@SeonJK Hi!

EuPI's simple example is here [link](https://github.com/euphony-io/euphony-listener/blob/development/app/src/main/java/co/jbear/euphony_listener/fragments/EuPICallFragment.java)

### EuPI Initialize
```kt
val rxManager: EuRxManager() = EuRxManager(EuOption.ModeType.EUPI)

rxManager.setOnWaveKeyPressed(18500) {  // 18500.0 -> 18500
  Toast.makeText(this, "menu request detected", Toast.LENGTH_SHORT).show()
}
```

### EuPI Listen
```kt
// listen : commonly this `listen` code should be after `setOnWaveKeyPressed`, `setOnWaveKeyDown`, `setOnWaveKeyUp`.
if (rxManager.listen()) {
  Log.d(TAG, "rxManager listen return : true")
} else {
  Log.d(TAG, "rxManager listen return : false")
}
```

### EuPI finish
```kt
rxManager.finish()
```

Let me know, if there is some issue :)

**Reply by SeonJK** [reply]
Oh, it works!! Thanks for help :)
